### ISSUE TITLE
Regular Expression Search

### ISSUE THREAD

**Issue by 8ullyMaguire** [bitmagnet-io/bitmagnet#227]
Implement support for using regular expressions in search queries to enable advanced pattern matching. This would cater to power users who prefer the flexibility and expressiveness of regular expressions in their searches.

**Comment by mgdigital** [comment]
Once the custom workflows PR is merged it will be possible to achieve a similar effect by adding tags at indexing time, based on regexes and many other criteria (although this isn't exactly regex search). As with other search enhancements they are certainly nice to have but the trade offs will be disk space of indexes and performance - leaving open for further discussion after custom workflows are implemented.
